### PR TITLE
[FIX] hr_timesheet_sheet_attendance: permissions

### PR DESCRIPTION
**Pull request by chienandalu**
Manual attendance users (`hr_attendance.group_hr_attendance`) aren't
supposed to be able to modify their own attendances times. So we raise
the permission to Attendance Officer.

cc @Tecnativa TT30141

please review @pedrobaeza @Yajo

### CODE REVIEW

File: hr_timesheet_sheet_attendance/views/hr_timesheet_sheet_view.xml [hunk 1]
```diff
@@ -55,7 +55,7 @@
                                 <button name="attendance_action_change" attrs="{'invisible': ['|', ('attendance_state','=',False), ('attendance_state', '=', 'checked_out')]}" type="object" string="Check Out"  style="margin-left: 20px"/>
                             </div>
                         </group>
-                        <field name="attendances_ids" nolabel="1" context="{'default_employee_id': employee_id}" colspan="4">
+                        <field name="attendances_ids" nolabel="1" context="{'default_employee_id': employee_id}" colspan="4" groups="hr_attendance.group_hr_attendance_user">
```
Comment (pedrobaeza) [inline hr_timesheet_sheet_attendance/views/hr_timesheet_sheet_view.xml]: Maybe we should avoid the editable thing, not to hide this.

Comment (chienandalu) [reply hr_timesheet_sheet_attendance/views/hr_timesheet_sheet_view.xml]: But the manual attendance user could still edit their attendances from the attendance line form popup

Comment (pedrobaeza) [reply hr_timesheet_sheet_attendance/views/hr_timesheet_sheet_view.xml]: You should prevent that with no_open or readonly=1

Comment (chienandalu) [reply hr_timesheet_sheet_attendance/views/hr_timesheet_sheet_view.xml]: Ok, changes done

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@bizzappdev can you please check?

File: hr_timesheet_sheet_attendance/views/hr_timesheet_sheet_view.xml [hunk 2]
```diff
@@ -55,8 +55,8 @@
                                 <button name="attendance_action_change" attrs="{'invisible': ['|', ('attendance_state','=',False), ('attendance_state', '=', 'checked_out')]}" type="object" string="Check Out"  style="margin-left: 20px"/>
                             </div>
                         </group>
-                        <field name="attendances_ids" nolabel="1" context="{'default_employee_id': employee_id}" colspan="4">
-                            <tree editable="bottom" create="false">
+                        <field name="attendances_ids" nolabel="1" options="{'no_create': True, 'no_open': True}" context="{'default_employee_id': employee_id}" readonly="1">
```
Comment (bizzappdev) [inline hr_timesheet_sheet_attendance/views/hr_timesheet_sheet_view.xml:58]: if it is readonly then why do we need 'no_create': True.
and why do we need to make it read-only? from the sheet normal user can adjust it manually.
especially in the case of the daily sheet. and someone forgets yesterday's signout.
correct me if I am wrong or misunderstood.

Comment (pedrobaeza) [reply hr_timesheet_sheet_attendance/views/hr_timesheet_sheet_view.xml:58]: About the readonly, the question is that Odoo doesn't allow to change attendances (not listing them) if your permission is "Manual attendance". This is a leak for allowing it.

Comment (bizzappdev) [reply hr_timesheet_sheet_attendance/views/hr_timesheet_sheet_view.xml:58]: True Menu access is not there for the Manual attendance group.

**Review by bizzappdev — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Let's merge it manually as the problem is not related to this change and we need to cover the leak.

**Pull request merged**